RestaurantsNearMe/Data/YelpAPIRequest.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  YelpAPIRequest.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2024-01-01.
+//
+
+import Foundation
+import Networking
+import Combine
+import CoreLocation
+
+class YelpAPIRequest<B: URLRequestBuilding>: YelpAPIRequesting {
+  private let requestBuilder: B
+  private let cloudKitServiceProvider: any CloudKitServiceProviding
+  private var activeLocation: CLLocation?
+  private var cancellables = Set<AnyCancellable>()
+  
+  init(
+    requestBuilder: B,
+    locationProvider: some LocationProviding,
+    cloudKitServiceProvider: some CloudKitServiceProviding
+  ) {
+    self.requestBuilder = requestBuilder
+    self.cloudKitServiceProvider = cloudKitServiceProvider
+    locationProvider
+      .currentLocation
+      .sink { self.activeLocation = $0 }
+      .store(in: &cancellables)
+  }
+  
+  func updateAPIKey() {
+    guard let privateKey = cloudKitServiceProvider.apiKey.value else {
+      return
```

Comment:
What happens if this process fails here?

Reply:
I wanted to cut off here before having "too much" in the PR but i'll probably throw an error that propagates to the UI.

Though in this case not having the `API` key is more of a business logic issue over a user issue, since it's being fetched internally via `CloudKit`